spec/forms/providers/means/state_benefit_form_spec.rb

```diff
@@ -0,0 +1,97 @@
+require "rails_helper"
+
+RSpec.describe Providers::Means::StateBenefitForm do
+  subject(:state_benefit_form) { described_class.new(params) }
+
+  let(:legal_aid_application) { create(:legal_aid_application) }
+  let(:transaction_type) { create(:transaction_type, :benefits) }
+  let(:params) do
+    {
+      description:,
+      amount:,
+      frequency:,
+      transaction_type_id: transaction_type.id,
+    }
+  end
+  let(:description) { "New State Benefit" }
+  let(:amount) { 123.00 }
+  let(:frequency) { "weekly" }
+
+  describe "#new" do
+    subject(:state_benefit_form) { described_class.new(model:) }
+
+    context "when there is an existing transaction" do
+      let(:model) { create(:regular_transaction, transaction_type:, legal_aid_application:, description: "Test state benefit") }
+
+      it "populates the form" do
+        expect(state_benefit_form).to have_attributes(transaction_type_id: transaction_type.id,
+                                                      description: "Test state benefit",
+                                                      amount: model.amount,
+                                                      frequency: model.frequency)
+      end
+    end
+
+    context "when no previous record exists" do
+      let(:model) { RegularTransaction.new(legal_aid_application:, transaction_type_id: transaction_type.id) }
+
+      it "creates an unpopulated form" do
+        expect(state_benefit_form).to have_attributes(transaction_type_id: transaction_type.id,
+                                                      description: nil,
+                                                      amount: nil,
+                                                      frequency: nil)
+      end
+    end
+  end
+
+  describe "#validate" do
+    subject(:validation) { state_benefit_form.valid? }
```

Comment:
nit, you can use `validate` to trigger validations to and that matches the #validate description, but either way

```suggestion
    subject(:validation) { state_benefit_form.validate }
```